name: 'ci'
on: [push, pull_request]      - name: Discord Webhook (Success)
        if: ${{success()}}
        run: |
          wget https://raw.githubusercontent.com/chronoDave/user-scripts/master/webhooks/github_discord.js
          node ./github_discord.js ${{secrets.WEBHOOK_URL}} success
      - name: Discord Webhook (Failure)
        if: ${{failure()}}
        run: |
          wget https://raw.githubusercontent.com/chronoDave/user-scripts/master/webhooks/github_discord.js
          node ./github_discord.js ${{secrets.WEBHOOK_URL}} error
jobs:
  ci:
    name: Lint & Test
    runs-on: ubuntu-latest
    env:
      WEBHOOK_FOOTER: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2.3.4
      - name: Get cache path
        id: yarn-cache-dir-path
        run: echo "::set-output name=dir::$(yarn cache dir)" 
      - name: Cache node_modules
        uses: actions/cache@v2.1.3
        id: yarn-cache
        with:
          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            yarn-
      - name: Install node
        uses: actions/setup-node@v2.1.3
      - name: Install packages
        run: yarn --frozen-lockfile
      - name: Lint
        run: yarn lint
      - name: Test
        run: yarn test
      - name: Discord Webhook (Success)
        if: ${{success()}}
        run: |
          wget https://raw.githubusercontent.com/chronoDave/user-scripts/master/webhooks/github_discord.js
          node ./github_discord.js ${{secrets.WEBHOOK_URL}} success
      - name: Discord Webhook (Failure)
        if: ${{failure()}}
        run: |
          wget https://raw.githubusercontent.com/chronoDave/user-scripts/master/webhooks/github_discord.js
          node ./github_discord.js ${{secrets.WEBHOOK_URL}} error
